### PR TITLE
Infer Nullity

### LINKED ISSUE #11886: Add null ability annotations to public API for kotlin
To improve using kotlin when integrating our SDK, we should annotate all public non-void api methods with nullability annotations as `@NonNull` and `@Nullable`. Kotlin uses these annotations to enforce null safety.

### PR DESCRIPTION
**Pull request by tobrun**
Closes #11886 , This PR improves consuming our library from kotlin by applying as much as possible `@Nullable/@NonNull` annotations. Most changes were performed automatically using `Analyze > Infer Nullity...` in Android Studio.

### CODE REVIEW

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java [hunk 1]
```diff
@@ -112,8 +117,8 @@ public void run() {
   }
 
   @UiThread
-  final void easeCamera(MapboxMap mapboxMap, CameraUpdate update, int durationMs, boolean easingInterpolator,
-                        final MapboxMap.CancelableCallback callback, boolean isDismissable) {
+  final void easeCamera(@NonNull MapboxMap mapboxMap, CameraUpdate update, int durationMs, boolean easingInterpolator,
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java]: `update` should not be null

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/renderer/textureview/TextureViewRenderThread.java [hunk 2]
```diff
@@ -118,7 +122,7 @@ void requestRender() {
   /**
    * May be called from any thread
    */
-  void queueEvent(Runnable runnable) {
+  void queueEvent(@Nullable Runnable runnable) {
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/renderer/textureview/TextureViewRenderThread.java]: `runnable` should not be null

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/snapshotter/MapSnapshotter.java [hunk 3]
```diff
@@ -338,21 +349,22 @@ private AttributionMeasure getAttributionMeasure(MapSnapshot mapSnapshot, Bitmap
...
     Bitmap selectedLogo = placement.getLogo();
     if (selectedLogo != null) {
       canvas.drawBitmap(selectedLogo, margin, snapshot.getHeight() - selectedLogo.getHeight() - margin, null);
     }
   }
 
-  private void drawAttribution(MapSnapshot mapSnapshot, Canvas canvas,
-                               AttributionMeasure measure, AttributionLayout layout) {
+  private void drawAttribution(@NonNull MapSnapshot mapSnapshot, @NonNull Canvas canvas,
+                               @NonNull AttributionMeasure measure, AttributionLayout layout) {
```
Comment (LukasPaczos) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/snapshotter/MapSnapshotter.java]: `layout` should not be null

**Comment by tobrun** [comment]
@LukasPaczos comments have been adressed, except https://github.com/mapbox/mapbox-gl-native/pull/13071#pullrequestreview-164290654:

```
//MapView.java
  public void addOnMapChangedListener(@NonNull OnMapChangedListener listener) {
  }
  public void removeOnMapChangedListener(@NonNull OnMapChangedListener listener) {
  }
```

**Pull request merged**